buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -408,7 +407,17 @@ public int numDirectArenas() {
      * Return the number of thread local caches used by this {@link PooledByteBufAllocator}.
      */
     public int numThreadLocalCaches() {
-        return threadCache.caches.get();
+        PoolArena<?>[] arenas = heapArenas != null ? heapArenas : directArenas;
```

Comment:
hmm I may miss something but it seems like you only either take heapArenas or directArenas into account. I think you need to check for both and run the for loop for both of them. no ?